pallets/Polkadex_Ido/src/lib.rs

```diff
@@ -514,4 +585,14 @@ impl<T: Config> Module<T> {
     pub fn round_account_id(hash: T::Hash) -> T::AccountId {
         T::ModuleId::get().into_sub_account(hash)
     }
+
+    fn incr_nonce() -> u128 {
+        let current_nonce: u128 = <Nonce>::get();
+        <Nonce>::put(current_nonce.saturating_add(1));
```

Comment:
A saturating add here will at some point always return the same value which defeats the presuppose of the nonce.
a wrap around is better for this kind of calculations.